pkg/tso/tso.go

```diff
@@ -169,36 +169,33 @@ func (t *timestampOracle) loadTimestamp() (time.Time, error) {
 			maxTSWindow = tsWindow
 		}
 	}
-	return maxTSWindow, nil
-}
-
-// save timestamp, if lastTs is 0, we think the timestamp doesn't exist, so create it,
-// otherwise, update it.
-func (t *timestampOracle) saveTimestamp(leadership *election.Leadership, ts time.Time) error {
-	key := t.getTimestampPath()
-	data := typeutil.Uint64ToBytes(uint64(ts.UnixNano()))
-	resp, err := leadership.LeaderTxn().
-		Then(clientv3.OpPut(key, string(data))).
-		Commit()
-	if err != nil {
-		return errs.ErrEtcdKVPut.Wrap(err).GenWithStackByCause()
-	}
-	if !resp.Succeeded {
-		return errs.ErrEtcdTxnConflict.FastGenByArgs()
-	}
-	t.lastSavedTime.Store(ts)
-	return nil
+	return maxTSWindow, resp.Kvs, nil
 }
 
 // SyncTimestamp is used to synchronize the timestamp.
-func (t *timestampOracle) SyncTimestamp(leadership *election.Leadership) error {
+func (t *timestampOracle) SyncTimestamp() error {
 	tsoCounter.WithLabelValues("sync", t.dcLocation).Inc()
 
 	failpoint.Inject("delaySyncTimestamp", func() {
 		time.Sleep(time.Second)
 	})
-
-	last, err := t.loadTimestamp()
+	// It is for compatibility with the old version and will try to load timestamp from the old path at the first time.
+	// Then, switch to the new path.
+	last, kvs, err := t.loadTimestamp()
+	if err != nil {
+		return err
+	}
+	// If old path exists, we should switch to the new path.
```

Comment:
We still need a configuration to turn on/off this path format change. Without this configuration, once this change is deployed to the prod, it will switch to new format even before the entire mcs feature is ready, which results in breaking change.

Reply:
"If old path exists, we should switch to the new path" -- this will become the mysterious logic in the code. Would it be better if we can control the behavior with explicit hot configuration? Probably we need introduce bi-directional smooth switching.

Reply:
I think we need a discussion about it. I'm not sure whether we should support downgrade operations.